pkg/code-generator/codegen/templates/resource_template.go

```diff
@@ -200,14 +200,27 @@ func (o *{{ .Name }}) DeepCopyObject() runtime.Object {
 {{- $crdGroupName = .Project.ProjectConfig.CrdGroupOverride }}
 {{- end}}
 
-var {{ .Name }}Crd = crd.NewCrd("{{ $crdGroupName }}",
-	"{{ lowercase (upper_camel .PluralName) }}",
-	"{{ $crdGroupName }}",
-	"{{ .Project.ProjectConfig.Version }}",
-	"{{ .Name }}",
-	"{{ .ShortName }}",
-	{{ .ClusterScoped }},
-	&{{ .Name }}{})
+var (
+	{{ .Name }}GVK = schema.GroupVersionKind{
+		Version: "{{ .Project.ProjectConfig.Version }}",
+		Group: "{{ $crdGroupName }}",
+		Kind: "{{ .Name }}",
+	}
+	{{ .Name }}Crd = crd.NewCrd(
+		"{{ lowercase (upper_camel .PluralName) }}",
+		{{ .Name }}GVK.Group,
+		{{ .Name }}GVK.Version,
+		{{ .Name }}GVK.Kind,
+		"{{ .ShortName }}",
+		{{ .ClusterScoped }},
+		&{{ .Name }}{})
+)
+
+func init() {
+	if err := crd.GetRegistry().AddCrd({{ .Name }}Crd); err != nil {
```

Comment:
this appears to be the only place the Registry gets referenced from the outside
i think we should simply provide a package-level `crd.AddCrd` and make everything else private